sklearnex/basic_statistics/tests/test_incremental_basic_statistics.py

```diff
@@ -272,15 +272,15 @@ def test_fit_single_option_on_random_data(
     X = X.astype(dtype=dtype)
     X_df = _convert_to_dataframe(X, sycl_queue=queue, target_df=dataframe)
     if weighted:
-        weights = gen.uniform(low=-0.5, high=+1.0, size=row_count)
+        weights = gen.uniform(low=0.0, high=1.0, size=row_count)
```

Comment:
what is the reason of this change? does it not work with negative weights?

Reply:
Other than being very very unusual. I guess what we should do is add a test which inverts the weights and tests to see if it matches min to max, etc etc.

Reply:
Yep, exactly, I guess in my experience before I had only interacted with circumstances where positive weight values were necessary. I guess if I wanted to normalize against another parameter/ calculate the moment of data using the various statistics which weren't positively constrained.  Btw, why was it using +1.0 vs just 1.0? Just seems stylistically different.

Reply:
I'll revert those changes, just let me know.

Reply:
I would keep negative weights in tests. I'm pretty sure that such applications exist (maybe they are not as widespread). And if our implementation supports negative weights, then imo better to test with negatives as well. 

Regarding +1.0, I probably just copied that from batch basic stat test, I agree that it's better to remove.